io-sim/test/Test/STM.hs

```diff
@@ -451,6 +476,8 @@ execTerm env t =
         let e' = execExpr env e
         throwSTM =<< snapshotExecValue e'
 
+      Catch t1 t2 -> execTerm env t1 `catch` \(_ :: ImmValue) -> execTerm env t2
```

Comment:
Yogesh pointed out that this only tests catch's with total handlers. The terms we generate currently do not have a way to express values of different types, and so we cannot generate a partial handler. That means these tests are somewhat incomplete.

Reply:
One option: write a separate test dedicated to checking that partial handlers work as expected.

Reply:
And then there's the additional gap where the exception value isn't being used in the handler. I had completely overlooked that, just focused on the type-level totality :)

That'd be much more complicated to fixup, I assume, since it would require generating a well-formed lambda.

Another possible compromise: add polymorphic constant function, eg `isZero` which is `False` for `()`, `2`, etc but `True` for `isZero 0`. .... hmm, though I realize `Term` doesn't even have lexical variable occurrences. So I doubt this would be small enough to include in this PR.

Reply:
This is not addressed in this PR

Reply:
Could you please create an issue for this in this repo then?

Reply:
New issue created https://github.com/input-output-hk/io-sim/issues/36. I also started implmenting it.